server/Assignments_filler/sheetFiller.py

```diff
@@ -0,0 +1,216 @@
+        
+        #get user display name
+        query = (f"SELECT `realname` FROM drupal.realname where uid={uid}")
+        name = get_from_db(query)[0][0]
+        
+        #check user's rank
+        query = (f"SELECT `rid` FROM drupal.users_roles where uid={uid}")
+        rank = get_from_db(query)[0][0]
+
+        #if new mission, delete comment
+        #if name changed spot, get the old comment   
+        localComment = ""
+        if nameChanged:
+            pass
+        elif name in goodListOfNames:
+            index = goodListOfNames.index(name)
+            if index < (len(listOfComments)):
+                if listOfComments[index] == []:
+                    localComment = ""
+                else:
+                    localComment = listOfComments[index][0]
+            else:
+                pass
+
+        #if needed, add "recruit"
+        if rank == 4:
+            if localComment == "":
+                localComment = "Recruit"
+            elif "Recruit" in localComment:
+                pass
+            else:
+                localComment = "Recruit, " + localComment
+            
+        #if no roles were filled, use "Rifleman"
+        if subject == "(No subject)":
+            subject = "Rifleman"
+
+        data = [name, subject, localComment]
+        writeRange = TEAM_ASSINGMENT_TAB + 'B' + str(row) + ":D" + str(row)
+        writeSheet(service, data, writeRange, 3)
+        row = row + 1
+    
+    #delete leftovers
+    logging.info("Deleting leftovers...")
+    data = ["", "", ""]
+    checkRange = TEAM_ASSINGMENT_TAB + 'B' + str(row) + ":D" + str(row)
+    NextCell = readSheet(service, checkRange, 3)
+    while NextCell != "":
+        writeSheet(service, data, checkRange, 3)
+        row = row + 1
+        checkRange = TEAM_ASSINGMENT_TAB + 'B' + str(row) + ":D" + str(row)
+        NextCell = readSheet(service, checkRange, 3)
+    
+    logging.info("Successfully filled!")
+
+def main():
+    #Setting up the API service
+    creds = None
+    # The file token.json stores the user's access and refresh tokens, and is
+    # created automatically when the authorization flow completes for the first time.
+    if os.path.exists('token.json'):
+        creds = Credentials.from_authorized_user_file('token.json', SCOPES)
+    # If there are no (valid) credentials available, let the user log in.
+    if not creds or not creds.valid:
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            flow = InstalledAppFlow.from_client_secrets_file(
+                'credentials.json', SCOPES)
+            creds = flow.run_local_server(port=0)
+        # Save the credentials for the next run
+        with open('token.json', 'w') as token:
+            token.write(creds.to_json())
+
+    service = build('sheets', 'v4', credentials=creds, cache_discovery=False)
+    readRange = TEAM_ASSINGMENT_TAB+RANGE_READ
+    
+    while True:
+        checker(service)
```

Comment:
```suggestion
    try:
        checker(service)
    except (KeyboardInterrupt, SystemExit):
        return
```
To handle shutting down gracefully instead of throwing an error.